apps/server/src/mindmap/mindmap.service.ts

```diff
@@ -0,0 +1,92 @@
+import { Injectable, NotFoundException } from '@nestjs/common';
+import { CreateMindmapDto } from './dto/create-mindmap.dto';
+import { UpdateMindmapDto } from './dto/update-mindmap.dto';
+import { MindmapRepository } from './repository/mindmap.repository';
+import Mindmap from './schema/mindmap.schema';
+import { Types } from 'mongoose';
+import FileService from 'src/utils/files/file.service';
+import { Readable } from 'stream';
+
+@Injectable()
+export class MindmapService {
+  constructor(
+    private readonly repository: MindmapRepository,
+    private readonly fileService: FileService,
+  ) {}
+  async create(
+    createMindmapDto: CreateMindmapDto,
+    id: Types.ObjectId,
+    name: string,
+  ): Promise<Mindmap> {
+    const doc: any = await this.repository.create({
+      ...createMindmapDto,
+      userId: id,
+      name,
+      path: 'temp'
```

Comment:
@BULMA-OO lets get rid of this `temp` form everywhere and make use to actual path where the files will be stored.